pkg/capr/planner/agent.go

```diff
@@ -1,15 +1,381 @@
 package planner
 
 import (
+	"bytes"
+	"encoding/base64"
 	"fmt"
 	"sort"
+	"text/template"
 
 	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/rancher/rancher/pkg/settings"
 	"github.com/rancher/rancher/pkg/systemtemplate"
 )
 
+const ClusterAgentInitialCreationManifest = `
+apiVersion: v1
+kind: Secret
+metadata:
+  name: rancher-cluster-agent-genesis-helper
+  namespace: kube-system
+data:
+  "create-cluster-agent-if-not-exists.sh": {{.ScriptB64}}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: rancher-cluster-agent-genesis
+  namespace: kube-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: rancher-cluster-agent-genesis-binding
+  namespace: kube-system
+subjects:
+- kind: ServiceAccount
+  name: rancher-cluster-agent-genesis
+  namespace: kube-system
+roleRef:
+  kind: ClusterRole
+  name: rancher-cluster-agent-genesis-role
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: rancher-cluster-agent-genesis-role
+rules:
+- apiGroups:
+  - '*'
+  resources:
+  - '*'
+  verbs:
+  - '*'
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: rancher-cluster-agent-genesis-deploy-job
+  namespace: kube-system
+spec:
+  backoffLimit: 10
+  template:
+    metadata:
+       name: rancher-cluster-agent-genesis-deploy
+    spec:
+        tolerations:
+        - operator: Exists
+        hostNetwork: true
+        serviceAccountName: rancher-cluster-agent-genesis
+        containers:
+          - name: pod
+            image: {{.AgentImage}}
+            env:
+            - name: CATTLE_SERVER
+              value: {{.URL}}
+            - name: CATTLE_CA_CHECKSUM
+              value: {{.CAChecksum}}
+            - name: CATTLE_TOKEN
+              value: {{.Token}}
+            - name: CATTLE_AGENT_STRICT_VERIFY
+              value: "{{.StrictVerify}}"
+            command: ["/bin/sh"]
+            args: ["/helper/create-cluster-agent-if-not-exists.sh"]
+            volumeMounts:
+            - name: secret-volume
+              mountPath: /helper
+        volumes:
+          - name: secret-volume
+            secret:
+              secretName: rancher-cluster-agent-genesis-helper
+        restartPolicy: Never
+`
+
+const ApplyManifestIfNotExistsScript string = `#!/bin/sh
+
+if [ -z "$CATTLE_SERVER" ] || [ -z "$CATTLE_TOKEN" ]; then
+	exit 1
```

Comment:
Can we print some error messages to make it easier for troubleshooting when it is necessary?